### PR TITLE
PERF-#4862: Don't call `compute_sliced_len.remote` when `row_labels/col_labels == slice(None)`

### LINKED ISSUE #4862: PERF: don't call `compute_sliced_len.remote` when `row_labels/col_labels` == `slice(None)`
This should speedup `partition.mask` operation.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4862  
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition.py [hunk 1]
```diff
@@ -187,13 +187,21 @@ def mask(self, row_labels, col_labels):
         """
         new_obj = super().mask(row_labels, col_labels)
         if isinstance(row_labels, slice) and isinstance(self._length_cache, Future):
-            new_obj._length_cache = DaskWrapper.deploy(
-                compute_sliced_len, row_labels, self._length_cache
-            )
+            if row_labels == slice(None):
+                # more faster way
```
Comment (vnlitvinov) [inline modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition.py:191]:
```suggestion
                # fast path - full axis take
```
this is usually what it's called (`fast path`), plus a short explanation why

here and everywhere else

File: modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition.py [hunk 2]
```diff
@@ -187,13 +187,21 @@ def mask(self, row_labels, col_labels):
...
+            else:
+                new_obj._length_cache = DaskWrapper.deploy(
+                    compute_sliced_len, row_labels, self._length_cache
+                )
         if isinstance(col_labels, slice) and isinstance(self._width_cache, Future):
-            new_obj._width_cache = DaskWrapper.deploy(
-                compute_sliced_len, col_labels, self._width_cache
-            )
+            if col_labels == slice(None):
+                # more faster way
```
Comment (anmyachev) [inline modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition.py:199]:
```suggestion
                # fast path - full axis take
```

File: modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py [hunk 3]
```diff
@@ -228,15 +228,23 @@ def mask(self, row_labels, col_labels):
         if isinstance(row_labels, slice) and isinstance(
             self._length_cache, ObjectIDType
         ):
-            new_obj._length_cache = compute_sliced_len.remote(
-                row_labels, self._length_cache
-            )
+            if row_labels == slice(None):
+                # more faster way
```
Comment (anmyachev) [inline modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py:232]:
```suggestion
                # fast path - full axis take
```

File: modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py [hunk 4]
```diff
@@ -228,15 +228,23 @@ def mask(self, row_labels, col_labels):
...
+                    row_labels, self._length_cache
+                )
         if isinstance(col_labels, slice) and isinstance(
             self._width_cache, ObjectIDType
         ):
-            new_obj._width_cache = compute_sliced_len.remote(
-                col_labels, self._width_cache
-            )
+            if col_labels == slice(None):
+                # more faster way
```
Comment (anmyachev) [inline modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py:242]:
```suggestion
                # fast path - full axis take
```

**Review by YarShev — APPROVED** [review]

**Pull request merged**